etcdctl/command/member_commands.go

```diff
@@ -86,7 +86,10 @@ func actionMemberAdd(c *cli.Context) {
 	mAPI := mustNewMembersAPI(c)
 
 	url := args[1]
-	ctx, cancel := context.WithTimeout(context.Background(), client.DefaultRequestTimeout)
+	totalCtx, totalCancel := contextWithTotalTimeout(c)
+	defer totalCancel()
+
+	ctx, cancel := contextWithPerRequestTimeout(c, totalCtx)
```

Comment:
The `per request timeout` is used at https://github.com/coreos/etcd/blob/master/etcdctl/command/util.go#L251. It is the timeout for each HTTP request(ref: http://godoc.org/github.com/coreos/etcd/client#Config). When we call mAPI.Add, it may include several HTTP requests, so we cannot set `per request timeout` as its context.

It is fine that we just use totalCtx for all requests in the this function IMO.